.github/validation_samples/kaon_enhanced/config.py

```diff
@@ -0,0 +1,162 @@
+import os
+import sys
+
+from LDMX.Framework import ldmxcfg
+
+thisPassName = 'test'
+p=ldmxcfg.Process(thisPassName)
+
+p.termLogLevel = 0
+# This need to be set but has no meaning given p.totalEvents
+p.maxEvents = 1
+p.totalEvents = int(0.5*int(os.environ['LDMX_NUM_EVENTS']))
+p.run = int(os.environ['LDMX_RUN_NUMBER'])
+nElectrons = 1 
+beamEnergy = 8;  #in GeV      
+
+from LDMX.SimCore import generators as gen
+from LDMX.SimCore import bias_operators
+from LDMX.SimCore import kaon_physics
+from LDMX.SimCore import photonuclear_models as pn
+from LDMX.Biasing import ecal
+from LDMX.Biasing import filters
+from LDMX.Biasing import particle_filter
+from LDMX.Biasing import util
+from LDMX.Biasing import include as includeBiasing
+
+if beamEnergy == 4 :
+     detector = 'ldmx-det-v14'
+     generator=gen.single_4gev_e_upstream_tagger()
+     bias_factor = 450.
+     bias_treshold = 2500.
+else :
+     detector = 'ldmx-det-v14-8gev'
+     generator=gen.single_8gev_e_upstream_tagger()
+     bias_factor = 550.
+     bias_treshold = 5000.
+
+mySim = ecal.photo_nuclear(detector, generator)
+mySim.description = f'{beamEnergy} GeV ECal Kaon PN simulation, xsec bias {bias_factor}' 
+mySim.biasing_operators = [ bias_operators.PhotoNuclear('ecal',bias_factor,bias_treshold,only_children_of_primary = True) ]
+
+# Configure the sequence in which user actions should be called.
+includeBiasing.library()
+mySim.actions.clear()
+mySim.actions.extend([
+        filters.TaggerVetoFilter(thresh=2*3800.),
+        # Only consider events where a hard brem occurs
+        filters.TargetBremFilter(recoil_max_p = 2*1500.,brem_min_e = 2*2500.),
+        # Only consider events where a PN reaction happnes in the ECal
+        filters.EcalProcessFilter(),
+        # Tag all photo-nuclear tracks to persist them to the event.
+        util.TrackProcessFilter.photo_nuclear()
+])
+
+# set up upKaon parameters
+mySim.kaon_parameters = kaon_physics.KaonPhysics.upKaons()
+
+# Alternative pn models
+myModel = pn.BertiniAtLeastNProductsModel.kaon() 
+# These are the default values of the parameters
+myModel.hard_particle_threshold=0. # Count particles with >= 200 MeV as "hard"
```

Comment:
```suggestion
myModel.hard_particle_threshold=0. # Count all (not stopped) particles as "hard"
```

Reply:
What is the reason for having no threshold? Or, i guess, what would the concern be with a 30 MeV kaon?

Reply:
This is the config used in the official production, 
https://github.com/LDMX-Software/LDCS/blob/main/productionScripts/v3.3/ldmxjob_kaon_v3.3.3.py.tpl#L66
Maybe @danyi211 remembers why zero was put here

Reply:
I think the reason is we want to accept all the kaons, including the soft ones that are more difficult to reject. According to [here](https://github.com/LDMX-Software/ldmx-sw/blob/9088c6798683f0aaaec727ba3b9e5dc69e07c7db/SimCore/src/SimCore/PhotoNuclearModels/BertiniAtLeastNProductsModel.cxx#L14), the kaons with energy less than this threshold would not be accepted.

Reply:
I think "difficult" events will include the ones whose final-state particles only have kaons and invisible particles...?

Reply:
ok so my take is that this region deserves some extra attention, and it's better to be more inclusive now (i.e. no threshold) and cut those out at the analysis level. And then given that this was the config used in production, I'd say we keep it as is, figure these out, and if needed, modify in the future (also meaning a future production). Does that sounds reasonable @EinarElen ?

Reply:
Yep, definitely not something to resolve here. I was mostly wondering what I was missing :)

Reply:
Done in the next push